azurelinuxagent/ga/monitor.py

```diff
@@ -122,22 +133,10 @@ def collect_and_send_events(self):
         """
         Periodically send any events located in the events folder
         """
-        try:
-            if self.last_event_collection is None:
-                self.last_event_collection = datetime.datetime.utcnow() - MonitorHandler.EVENT_COLLECTION_PERIOD
+        event_list = collect_events()
 
-            if datetime.datetime.utcnow() >= (self.last_event_collection + MonitorHandler.EVENT_COLLECTION_PERIOD):
-                try:
-                    event_list = collect_events()
-
-                    if len(event_list.events) > 0:
-                        self.protocol.report_event(event_list)
```

Comment:
Initially we were reporting errors everytime it happened if unable to send telemetry (i.e. every minute). Now we will only log once per hour as per PeriodicOperation. Should we revisit that?
(Also added another concern with the way PeriodicOperation logs in the PeriodicOperation file)

Reply:
(I'll address your 2 comments here)

While refactoring the code I realized we were always logging on error. Given the frequency of some of those operations this has the potential of producing very large logs. I agree that we may be hiding errors if they are different but our current periodic logging has issues of its own.

To be honest, I've never seen those operations fail, so as far as this PR for release 2.2.48 I see 2 options: revert to the previous behavior of always logging or use this simple approach (which could be improved in subsequent PRs if one of us feels it is needed). I am inclined towards the latter, but can revert if you prefer so.